### DISCUSSION TITLE
Terrain And Objects

### DISCUSSION THREAD

**Discussion by GreedyDragon81** [D2klaas/Godot-4-VectorFieldNavigation#3] [Q&A]
Would this work with a terrain generator like terrain3D?

If I'm making an rts game, and I place some buildings and walls, with this recalculate the field with those nodes counted as a penalty?

**Comment by D2klaas** [comment]
Hi,
this pathfinding addon is fully indipendent from your terrain and scene. You just have to provide a dataset(field) which reflects your scenery. Its the same as when using astar algorythm. The penalty fields are just a convinient way to organise your movement penylties. It also provides a smart way to make dynamic changes to the field.

**Reply by GreedyDragon81** [reply]
Ok cool I was able to extract a height map from terrain3D. 

So since it's dynamic, does that mean as I add buildings and I want to "re bake" the vector field, the buildings will be included and the units should avoid it? Forgive me I am new to vector flow fields.

**Reply by D2klaas** [reply]
Yes, you can add a field (most likly static boolean) for excluding the space the buildings are on. Plese read the the manual here ...
https://github.com/D2klaas/Godot-4-VectorFieldNavigation?tab=readme-ov-file#modfields

You should use a static modfield for this since the buildings will not change often i guess, so you benefit from caching.

Rebaking is a constant process since your targets will most likly move on the battlefield.

**Reply by GreedyDragon81** [reply]
Thanks!! Yes your documentation has been helpful. This is a really cool project. Thanks for your work on it!
